coalib/results/Diff.py

```diff
@@ -276,23 +282,36 @@ def split_diff(self, distance=1):
         # always.
         yield this_diff
 
-    def range(self, filename):
+    def range(self, filename=None):
         """
         Calculates a SourceRange spanning over the whole Diff. If something is
         added after the 0th line (i.e. before the first line) the first line
         will be included in the SourceRange.
 
         The range of an empty diff will only affect the filename:
 
+        >>> range = Diff([], filename="file").range()
+        >>> range.file is None
```

Comment:
I was merely following the existing  doctest without really thinking about it. How I understand the comment on line 291 is that the filename will change, so the simplest way to test it is to check if there is, in fact, a file connected to the range.